test/src/ics/ScalarUOIC.C

```diff
@@ -0,0 +1,41 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ScalarUOIC.h"
+#include "MTUserObject.h"
+
+registerMooseObject("MooseTestApp", ScalarUOIC);
+
+template <>
+InputParameters
+validParams<ScalarUOIC>()
+{
+  InputParameters params = validParams<ScalarInitialCondition>();
+  params.addRequiredParam<UserObjectName>("user_object",
+                                          "The MTUserObject to be coupled into this IC");
+  return params;
+}
+
+ScalarUOIC::ScalarUOIC(const InputParameters & parameters)
+  : ScalarInitialCondition(parameters)
+{
+}
+
+void
+ScalarUOIC::initialSetup()
+{
+  // at the time of creation, user objects are not available, so need to wait with this
+  _uo = &getUserObject<MTUserObject>("user_object");
```

Comment:
Really? The normal ICs can retrieve UserObjects in their constructor. Somethings not right here:
https://github.com/idaholab/moose/blob/next/test/src/ics/ElementUOIC.C

Reply:
https://github.com/idaholab/moose/blob/next/framework/src/base/Moose.C#L270-L274

Reply:
You are right, this should work. I confused myself reading `SolutionFunction`.